### PR TITLE
Update workers to have more easily updated user setting, split out user recipe and consolidate worker types

### PR DESCRIPTION
**Pull request by JoeNunnelley**
converged all recipes and validated that the resource (ruby, python, sbt, etc) were configured correctly.

### CODE REVIEW

**Comment by JoeNunnelley** [comment]
@michaelb990 can you take a look at this. Just realized it needed to be assigned.

File: recipes/_user.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+iss_db_credentials = {}
+
+node.run_state[:jenkins_private_key] = private_deployment_key
+
+user a_user do 
+  comment "user #{a_user}"
+  home a_user_home
+  shell '/bin/bash'
+  action :create
+  not_if "getent passwd #{a_user}"
```
Comment (michaelb990) [inline recipes/_user.rb]: doesn't the `user` resource already have a way of converging this intelligently?

why do you need this?

File: recipes/_user.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  variables({
+    :github_user => github['username'],
+    :github_access_token => github['access_token']
+  })
+end
+
+# Install general required packages
+%w(
+  git
+  vim
```
Comment (michaelb990) [inline recipes/_user.rb]: +emacs

File: recipes/worker-cheetah.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 worker_user = node['worker_user']
-user_home = "/home/#{worker_user}"
+user_home = "/var/lib/#{worker_user}"
```
Comment (michaelb990) [inline recipes/worker-cheetah.rb]: why are you changing this?

File: recipes/_clortho.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-user_home = "/home/#{node['worker_user']}"
+user_home = "/var/lib/#{node['worker_user']}"
```
Comment (michaelb990) [inline recipes/_clortho.rb]: this should just be a node attribute, but again, i don't understand why it is changing...

**Comment by JoeNunnelley** [comment]
I'm trying to get the soql postgres adapter cut job to run on a worker but its having trouble building on the worker because the its needs to have push permissions on github. when running on the server the user jenkins has those permissions, when running on a worker the user 'ubuntu' does NOT so the build fails. I'm trying to convert our workers to run as 'jenkins' and have the appropriate github credentials to push.

File: recipes/_user.rb [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+github = a_user_credentials['github']
+template "#{a_user_home}/.git-credentials" do
+  source 'git-credentials.erb'
+  action :create
+  owner a_user
+  group a_user
+  mode '0600'
+  variables({
+    :github_user => github['username'],
+    :github_access_token => github['access_token']
```
Comment (michaelb990) [inline recipes/_user.rb]: this should be added on launch, not when we bake an AMI.

**Comment by michaelb990** [comment]
@JoeNunnelley can we close this PR? Don't think we need it.

**Comment by JoeNunnelley** [comment]
@michaelb990 I've re-purposed this PR to break out the user config, collapse 4 worker types down to 2 and delete the extra.

File: metadata.rb [hunk 6]
```diff
@@ -4,12 +4,14 @@
 license          'All rights reserved'
 description      'Installs/Configures socrata-jenkins'
 long_description 'Installs/Configures socrata-jenkins'
-version          '0.2.2'
+version          '0.2.3'
```
Comment (michaelb990) [inline metadata.rb]: bump minor version.

File: recipes/_user.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+directory user_home do
+  user a_user
+  owner a_user
+  group a_user
+  mode '0755'
+  action :create
+end
+
+# Ensure data directory is present, required because jenkins_data is /data/jenkins
+directory '/data'
```
Comment (michaelb990) [inline recipes/_user.rb]: The jenkins workers don't have data volumes and the `Remote FS root` is set to `/home/ubuntu/jenkins`. I don't think we need this.

File: metadata.rb (hunk 6 above)
Comment (JoeNunnelley) [reply metadata.rb]: fixed

**Comment by JoeNunnelley** [comment]
why this PR? here are the reasons for the changes included herein.

1.) move user provisioning to a separate recipe to make testing it in isolation easier.
2.) remove the unused worker.rb recipe that is not used by anything anymore.
3.) merge the cheetah specific recipe in with the others for a more general worker solution while retaining the pg94/pg96 split (trying to make them live together is unecessarily complex). Now we only have to provision 2 worker types on the server.
4.) reduction of ruby versions installed so that we now adhere to a policy of n and n-1 versions only.  reduces build times.
5.) fix go language path resolutions issue.
6.) move package installs to more appropriate locations. apt installs in _common, gem installs in _ruby

File: recipes/_user.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+a_user = node['worker_user']
```
Comment (michaelb990) [inline recipes/_user.rb]: `a_user`=>`worker_user`

**Pull request merged**